### ISSUE TITLE
lazy session update

### ISSUE THREAD

**Issue by rafaelcardoso** [kcbanner/connect-mongo#152]
Guys, do you have interest in an option on the instantiation moment of this middleware that allow us to lazy update the session database?

I had this idea because every single time that I refresh the page ´express-session´ just call the method req.session.touch(), and this call MongoStore.prototype.touch, that make and update on mongoDB, even if I set resave:false and do NOT change anything on the session data.

I guess that it would be nice because we can say to the touch() method that we want to update the session once a day, or once on each 12 hours, extending the lifetime of our session without doing massive updates on database, I guess that this will give to us a large gain in performance.

What do you think?

**Comment by rafaelcardoso** [comment]
Here is my attempt to propose this at the express-session repo ):
https://github.com/expressjs/session/issues/113

**Comment by rafaelcardoso** [comment]
Here is how this could be implemented:

On the moment of instantiation of connect-mongo:

```javascript
new mongoStore({
 updateAfter: 86400000 // one day
})
```

```javascript
// inside MongoStore.prototype.touch 
var timeElapsed = Date.now() - session.cookie._expires;

if(this.options.updateAfter && timeElapsed < this.options.updateAfter){
 callback();
}
```

pretty simple, right?

**Comment by rafaelcardoso** [comment]
Here is how I implemented this feature https://github.com/rafaelcardoso/connect-mongo/commit/d07f5c88ae64e72acd6f9056c5cfaefc2cf4291b

To use this, you just need to pass an property called `updateAfter` on instance function, like this

```javascript
app.use(session({
	secret: 'keyboard cat'
	cookie: {
		maxAge: 3 * 24 * 3600000
	},
	saveUninitialized: false,
	resave: false,
	rolling: false,
	store: new mongoStore({
		updateAfter: 24 * 3600000, // update only once a day
		serialize: function (session) {
			var obj = {};
			obj.passport = session.passport;
			obj.lastModified = Date.now();
			obj.cookie = session.cookie.toJSON ? session.cookie.toJSON() : session.cookie;
			obj = JSON.stringify(obj);
			return obj;
		},
		unserialize: function (session) {
			return JSON.parse(session);
		}
	})
}));
```

it's pretty simple, right?
What do you think?

**Comment by jdesboeufs** [comment]
Interesting but very advanced feature!
Maybe you should prefer `touchAfter` which seems more consistent.

To release I need tests and documentation ;)

**Comment by jdesboeufs** [comment]
Fixed with #154.
